src/tests/class_tests/openms_gui/source/GUI/TVPrefDialog_test.cpp

```diff
@@ -0,0 +1,352 @@
+  QVERIFY2(UI->feature_icon_size_2D, "'feature icon size' spin box not created.");
+  QVERIFY2(UI->label_13, "'Peak gradient:' label not created.");
+  QVERIFY2(UI->label_17, "'m/z axis:' label not created.");
+  QVERIFY2(UI->label_21, "'feature icon:' label not created.");
+  QVERIFY2(UI->label_22, "'feature icon size:' label not created.");
+
+  // 3D tab
+  QVERIFY2(UI->peak_3D, "Peak 3D multi gradient selector not created.");
+  QVERIFY2(UI->shade_3D, "Shade mode combo box not created.");
+  QVERIFY2(UI->line_width_3D, "Line width spin box not created.");
+  QVERIFY2(UI->label_18, "'Peak colors:' label not created.");
+  QVERIFY2(UI->label_19, "'Shade mode:' label not created.");
+  QVERIFY2(UI->label_20, "'Line width:' label not created.");
+
+  // TSG tab
+  QVERIFY2(UI->param_editor_spec_gen_, "Parameter editor not created.");
+  QVERIFY2(UI->tolerance, "Tolerance double spin box not created.");
+  QVERIFY2(UI->unit, "Unit combo box not created.");
+  QVERIFY2(UI->label_2, "'Alignment Tolerance:' label not created.");
+}
+
+void TestTVPrefDialog::testGui()
+{
+  // TODO: currently Qt::ComboBox, OpenMS::MultiGradientSelector and OpenMS::ParamEditor are not tested!
+
+  enum
+  {
+    GENERAL,
+    ONE_D,
+    TWO_D,
+    THREE_D,
+    TSG
+  };
+
+  QTabBar* tab_bar = UI->tabWidget->tabBar();
+
+  /////////////////////////////////////////////
+  //             'General' tab               //
+  /////////////////////////////////////////////
+
+  // click on tab
+  QTest::mouseClick(tab_bar, Qt::LeftButton, 0, tab_bar->tabRect(GENERAL).center());
+  QTest::qWait(DELAY);
+
+  // test path input methods
+  
+  // file dialog
+  QTimer::singleShot(DELAY, this, &TestTVPrefDialog::checkFileDialog_);
+  QTest::mouseClick(UI->browse_default, Qt::LeftButton, 0, QPoint());
+  QTest::qWait(DELAY);
+
+  // line edit
+  QTest::keyClicks(UI->default_path, "C:\dev");
+  QVERIFY2(UI->default_path->text() == "C:\dev", "Line edit for default path broken.");
+  QTest::qWait(DELAY);
+
+  // test check boxes
+  testCheckBox(UI->default_path_current);
+  testCheckBox(UI->use_cached_ms1);
+  testCheckBox(UI->use_cached_ms2);
+
+  // TODO: 'map_default' (Qt::ComboBox), 'map_cutoff' (Qt::ComboBox), 'on_file_change' (Qt::ComboBox)
+
+  /////////////////////////////////////////////
+  //             '1D view' tab               //
+  /////////////////////////////////////////////
+
+  // click on tab
+  QTest::mouseClick(tab_bar, Qt::LeftButton, 0, tab_bar->tabRect(ONE_D).center());
+  QTest::qWait(DELAY);
+
+  // test color selectors
+  QTimer::singleShot(DELAY, this, &TestTVPrefDialog::checkColorDialog_);
+  QTest::mouseClick(UI->color_1D, Qt::LeftButton, 0, QPoint());
+  QTest::qWait(DELAY);
+
+  QTimer::singleShot(DELAY, this, &TestTVPrefDialog::checkColorDialog_);
+  QTest::mouseClick(UI->selected_1D, Qt::LeftButton, 0, QPoint());
+  QTest::qWait(DELAY);
+
+  QTimer::singleShot(DELAY, this, &TestTVPrefDialog::checkColorDialog_);
```

Comment:
Why do you need to create a timed event here?
Could you not also 
1) click
2) checkColorDialog_ 
??

And maybe providing the widget to test as a parameter would be more explicit (on first sight, the code looks like it tests the same thing three times, but it does not...)

Reply:
This is needed because when a dialog is opened the tests stops being executed until the dialog is closed.

Reply:
ah,yes. 
Please add a comment, so other people know why its done this way without finding out the hard way